crates/storage/provider/src/traits/chain.rs

```diff
@@ -139,3 +139,42 @@ impl CanonStateNotification {
         receipts
     }
 }
+
+/// A type that allows to register finalized block related event subscriptions
+/// and get notified when a new finalized block header is available.
+pub trait FinalizedBlockHeaderSubscription: Send + Sync {
```

Comment:
let's combine safe and finalized into one trait

maybe we can come up with a name that isn't also tied to headers, essentially these are derived from forkchoice, so perhaps `Forkchoice(Block)Subscriptions`